Sources/Connection.swift

```diff
@@ -0,0 +1,20 @@
+//
+//  Connection.swift
+//  C7
+//
+//  Created by Scott Byrns on 4/1/16.
+//
+//
+
+public protocol Connection {
+    
+    associatedtype StatusType
+    
+    func open() throws
+    func close() -> Bool
+    
+    var connectionInfo: ConnectionInfo { get }
+    var status: StatusType { get }
+    var mostRecentError: ErrorProtocol? { get }
```

Comment:
Status I would like to leave up to the implementer of a given connection. All connections have a status. How do you feel about genetics?

Connection<ConnectionStatus> { var status: ConnectionStatus { get } }